### PR TITLE
ripgrep: 0.9.0 -> 0.10.0

### LINKED ISSUE #46350: ripgrep just released version 0.10.0
## Issue description

[ripgrep][] just release version [v0.10.0][] and I'd love to see the Nix package updated.

[ripgrep]: https://github.com/BurntSushi/ripgrep
[v0.10.0]: https://github.com/BurntSushi/ripgrep/releases/tag/0.10.0

/cc @tailhook

### PR DESCRIPTION
**Pull request by dywedir**
[0.10.0 (2018-09-07)](https://github.com/BurntSushi/ripgrep/blob/master/CHANGELOG.md#0100-2018-09-07)

> Major new features include PCRE2 support, multi-line search and a JSON output format.

closes https://github.com/NixOS/nixpkgs/issues/46350

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

cc @tailhook

### CODE REVIEW

File: pkgs/tools/text/ripgrep/default.nix [hunk 1]
```diff
@@ -1,19 +1,24 @@
-{ stdenv, fetchFromGitHub, rustPlatform, asciidoc, docbook_xsl, libxslt  }:
+{ stdenv, fetchFromGitHub, rustPlatform, asciidoc, docbook_xsl, libxslt
+, withPCRE2 ? false, pcre2 ? null
```
Comment (dywedir) [inline pkgs/tools/text/ripgrep/default.nix]: Maybe `withPcre` or `withpcre`?

Comment (xeji) [reply pkgs/tools/text/ripgrep/default.nix]: I think it's fine as is. Camel casing looks weird for abbreviations anyway.

**Comment by xeji** [comment]
let's check darwin build
@GrahamcOfBorg build ripgrep

**Comment by xeji** [comment]
Let's try darwin again. Unfortunately we cannot tell ofborg to only build one platform...
@GrahamcOfBorg build ripgrep

**Pull request merged**